core/src/main/scala/chisel3/Data.scala

```diff
@@ -805,6 +805,39 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
 
   /** A non-ambiguous name of this `Data` for use in generated Verilog names */
   def typeName: String = simpleClassName(this.getClass)
+
+  /** Extension point to customize [[Connectable]] waive behavior for this [[Data]].
+    *
+    * This method will be called on each recursive [[Data]] within this [[Data]], including itself, any nested
+    * aggregates, and leaf elements. If it returns true for a given [[Data]], that [[Data]] will be waived.
```

Comment:
I know you're describing behavior, not changing existing behavior, but does this mean that you can have multiple instances of this method applied? Like if you have nested Bundles, do you call this method for every parent Bundle on the way up? Seems expensive.

Reply:
I think so? This is me trying to describe the behavior of DataMirror.collectMembers. Concretely, this is where it is happening, and I'm just using these methods to override those three default parameters: https://github.com/chipsalliance/chisel/blob/84c0fbfba6eecc2e53808d5638135036376c9483/core/src/main/scala/chisel3/connectable/Connectable.scala#L149-L154

Reply:
See comment on the implicit conversion, I don't _think_ it does (but we should check if I'm right): https://github.com/chipsalliance/chisel/pull/3978#discussion_r1558121125